tests/api/circulation/test_request_actions.py

```diff
@@ -0,0 +1,307 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019 RERO
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Test item circulation request actions."""
+
+
+import pytest
+from invenio_circulation.errors import RecordCannotBeRequestedError
+from utils import item_record_to_a_specific_loan_state
+
+from rero_ils.modules.items.models import ItemStatus
+from rero_ils.modules.loans.api import LoanState
+
+
+def test_add_request_1_actions(client, librarian_martigny_no_email,
+                               item_lib_martigny, loc_public_martigny,
+                               patron_martigny_no_email, circulation_policies,
+                               patron2_martigny_no_email):
+    """Test add request_1 actions"""
+    # test add_request_1_1 action
+    # item on_shelf (no current loan): PENDING loan does not exist
+    params = {
+        'patron_pid': patron_martigny_no_email.pid,
+        'transaction_location_pid': loc_public_martigny.pid,
+        'transaction_user_pid': librarian_martigny_no_email.pid,
+        'pickup_location_pid': loc_public_martigny.pid
+    }
+
+    item, loan = item_record_to_a_specific_loan_state(
```

Comment:
in the `tests/api` directory we have only the REST API tests. If you need to test the python API please put into `tests/ui/*/*_api.py`